File: tests/java/TestObserver.java
public class TestObserver {
    public static void main(String[] args) {
        System.out.println("Testing Java Observer...");
        try {
            client.main(args);
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }
        System.out.println("Java Observer Test Passed!");
    }
}

File: tests/c/TestObserver.c
#define main pattern_main
#include "../../oop_in_c/Observer/Observer.c"
#undef main
#include <stdio.h>

int main() {
  printf("Testing C Observer...\n");
  // Call pattern main
  pattern_main();
  printf("C Observer Test Passed!\n");
  return 0;
}
